### PR TITLE
Allow GitHub usernames as port maintainers

### PR DESCRIPTION
**Pull request by neverpanic**
After the move to GitHub, it is sometimes difficult to figure out
whether a pull request was sent or a ticket was opened by a maintainer,
because both these systems use GitHub usernames now.

This leads to more confusion with developers and users than needed.
Change the documentation to allow GitHub usernames as maintainers for
Portfiles to avoid this.

Using a GitHub username leaves us with the problem that there is no
email address available for this user, unless the user has added
a public email address to their GitHub account. Email addresses are
instead available in commits, and maintainers can always be reached by
ticket or GitHub comment.

### CODE REVIEW

**Comment by neverpanic** [comment]
Let's discuss whether this approach would be feasible. Fortunately, it seems that we probably won't have to change base for this at all, unless we want to remove the leading `@` symbol from GitHub usernames, see macports/macports-base#3.

**Comment by jmroot** [comment]
I would actually lean towards listing both a github username and an email address. (But still allow just one if e.g. non-committers don't want to expose their personal email address.) That might require base changes to allow a syntax that unambiguously indicates that the two refer to the same person, though.

**Review by jmroot — APPROVED** [review]
This is at the very least a good start and should reduce confusion while we work things out.

**Comment by jdberry** [comment]
I'm a little uncomfortable with using @ as the prefix that introduces a GitHub user name, because it's so tied to email addresses (though I admit that twitter, GitHub, etc, usage has usurped this long ago).

I wonder instead about using the ~ to introduce a GitHub user name. This would also have the advantage that a single entry could specify both a user name and an email address, effectively tying them together in the port file:

Github user name alone: ~jdberry
short (macports) email address + Github: jberry~jdberry
fq email address + GitHub: jberry@macports.org~jdberry
obscured email address + GitHub: macports.org:jberry~jdberry

**Comment by jdberry** [comment]
And left unsaid, the idea that this syntax ties the different forms of address together in the port file means that we have additional meta information that we can use to display these to the end-user as a single individual, which could be a real benefit. i.e., that there's a single maintainer that can be reached in multiple ways, rather than two potentially separate individuals.

**Comment by jdberry** [comment]
I'll point out that ~ is technically legal in the local part of email addresses, though I'd argue that isn't something we probably have to worry about???

**Comment by jdberry** [comment]
Another option would be to overload the : that we already carve out, using a double colon to specify a GitHub address:

Github user name alone: ::jdberry
short (macports) email address + Github: jberry::jdberry
fq email address + GitHub: jberry@macports.org::jdberry
obscured email address + GitHub: macports.org:jberry::jdberry

**Review by larryv — APPROVED** [review]
Or we could use lists.

``` tcl
maintainers {@neverpanic cal} email.org:someone
```

But this looks good for now.

**Comment by jdberry** [comment]
Or we could use lists. Good point @larryv. We probably don't need to invent syntax to save a few characters when the language already gives us a perfectly good, and (relatively) well-known syntax for grouping.

**Comment by neverpanic** [comment]
I thought about the lists as well, I think that's the best solution. I'll come up with an implementation first and then change this PR to document it accordingly.

**Comment by neverpanic** [comment]
I implemented the list approach in macports/macports-base#3. Let's get that reviewed and merged and then I'll fix the docs.

**Comment by raimue** [comment]
I guess we could merge this now after this was finally fixed with the 2.4 release?

**Comment by neverpanic** [comment]
Absolutely. I think I wrote this before we settled on the `{}` syntax grouping a single person, though, so that would still have to be added.

**Comment by raimue** [comment]
Merged in 5d94c2a with slight editing (rebased and one whitespace fix).

**Comment by mojca** [comment]
I just realized that our guide almost never mentions github handles (search for `maintainer` and start at the top of our guide). We should fix those place to encourage new contributors to add the github handles, otherwise each PR is missing that info and it becomes impossible to track maintainership, esp. in cases where people submitting a PR might not even have an account on Trac.

**Pull request closed without merging**
